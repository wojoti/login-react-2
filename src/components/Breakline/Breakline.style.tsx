import styled from "styled-components";
export const WrapperA = styled.div`
  width: 100%;
`;
export const WrapperB = styled.div`
  height: 2px;
  width: 100%;
  background-color: rgb(229 231 235);
`;
export const WrapperC = styled.div`
  position: relative;
`;
export const Text = styled.p`
  border: 1px solid rgb(156 163 175);
  border-radius: 4px;
  position: absolute;
  background-color: white;
  font-size: 14px;
  color: rgb(156 163 175);
  top: 50%;
  left: 50%;
  padding: 1px 2px;
  transform: translateX(-50%) translateY(-50%);
`;
