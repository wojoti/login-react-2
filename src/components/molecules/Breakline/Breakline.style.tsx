import styled from "styled-components";
export const MaxWidthDiv = styled.div`
  width: 100%;
`;
export const GreyLineDiv = styled.div`
  height: 2px;
  width: 100%;
  background-color: rgb(229 231 235);
`;
export const RelativeDiv = styled.div`
  position: relative;
`;
export const StyledText = styled.p`
  border: 1px solid rgb(156 163 175);
  border-radius: 4px;
  position: absolute;
  background-color: white;
  font-size: 14px;
  color: rgb(156 163 175);
  top: 50%;
  left: 50%;
  padding: 1px 2px;
  transform: translateX(-50%) translateY(-50%);
`;
