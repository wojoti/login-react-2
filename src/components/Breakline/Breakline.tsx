import React from "react";
import { WrapperA, WrapperB, WrapperC, Text } from "./Breakline.style";

type Props = {
  children?: string;
};

const Button = (props: Props) => {
  return (
    <WrapperA>
      <WrapperB>
        <WrapperC>
          <Text>{props.children}</Text>
        </WrapperC>
      </WrapperB>
    </WrapperA>
  );
};

export default Button;
