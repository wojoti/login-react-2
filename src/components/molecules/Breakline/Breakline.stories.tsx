/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Breakline from "./Breakline";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Molecules/Breakline",
  component: Breakline,
} as ComponentMeta<typeof Breakline>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Breakline> = (args) => (
  <Breakline {...args} />
);

export const Example = Template.bind({});
Example.args = { children: "OR" };
