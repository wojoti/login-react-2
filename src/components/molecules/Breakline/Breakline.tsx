import * as Styles from "./Breakline.style";

export interface BreaklineProps {
  children?: string;
  testId?: string;
}

function Breakline({ children, testId }: BreaklineProps) {
  return (
    <Styles.MaxWidthDiv data-testid={testId}>
      <Styles.GreyLineDiv>
        <Styles.RelativeDiv>
          <Styles.StyledText>{children}</Styles.StyledText>
        </Styles.RelativeDiv>
      </Styles.GreyLineDiv>
    </Styles.MaxWidthDiv>
  );
}

export default Breakline;
