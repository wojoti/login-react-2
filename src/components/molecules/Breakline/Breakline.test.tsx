import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Breakline, { BreaklineProps } from "./Breakline";

const props: BreaklineProps = {
  children: "test breakline",
  testId: "test-breakline-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Breakline {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render breakline", () => {
  render(<Breakline {...props} />);
  const breaklineElement = screen.getByTestId("test-breakline-id");
  expect(breaklineElement).toBeInTheDocument();
});

test("should render breakline with text", () => {
  render(<Breakline {...props} />);
  const breaklineElement = screen.getByTestId("test-breakline-id");
  expect(breaklineElement).toBeInTheDocument();
  expect(breaklineElement).toHaveTextContent("test breakline");
});
