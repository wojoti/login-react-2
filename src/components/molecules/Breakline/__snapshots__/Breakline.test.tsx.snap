// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  height: 2px;
  width: 100%;
  background-color: rgb(229 231 235);
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  position: relative;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 1px solid rgb(156 163 175);
  border-radius: 4px;
  position: absolute;
  background-color: white;
  font-size: 14px;
  color: rgb(156 163 175);
  top: 50%;
  left: 50%;
  padding: 1px 2px;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

<div
  className="c0"
  data-testid="test-breakline-id"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <p
        className="c3"
      >
        test breakline
      </p>
    </div>
  </div>
</div>
`;
